lib/services/services_linux.c

```diff
@@ -706,8 +692,16 @@ services_os_action_execute(svc_action_t * op)
             crm_perror(LOG_ERR, "pipe() failed");
         }
 
-        set_fd_opts(sigchld_pipe[0], O_NONBLOCK);
-        set_fd_opts(sigchld_pipe[1], O_NONBLOCK);
+        rc = crm_set_nonblocking(sigchld_pipe[0]);
+        if (rc < 0) {
+            crm_warn("Could not set pipe input non-blocking: %s " CRM_XS " rc=%d",
+                     pcmk_strerror(rc), rc);
+        }
+        rc = crm_set_nonblocking(sigchld_pipe[1]);
+        if (rc < 0) {
+            crm_warn("Could not set pipe output non-blocking: %s " CRM_XS " rc=%d",
+                     pcmk_strerror(rc), rc);
+        }
```

Comment:
What does it actually mean if setting one of these pipes to non-blocking goes wrong?
Daemon blocking on the first child that is writing out something?
But of course it is the behavior we already had up to now ...

Reply:
Good question. I suspect it would never fail in practice, short of passing in a bad fd, so hopefully we can safely ignore it :)